src/Congo.WebApi/Controllers/SellerProductsController.cs

```diff
@@ -19,12 +20,12 @@ public SellerProductsController(IMediator mediator)
 
         [HttpPost]
         [Route("products")]
-        public async Task<ActionResult> Create(InsertProductRequest product)
+        public async Task<ActionResult<Guid>> Create(InsertProductRequest product)
         {
             var productId = await _mediator.Send(new InsertProductCommand
                 (product.Name, product.Description, product.Price, product.ImageUrl));
 
-            return Ok(productId);
```

Comment:
I think I miss-understood you here. I think its ok to use return Ok(product.Id); so long as we make sure that we declared that we are returning a Guid in the method signature

Reply:
When we specify the actionresult type explicitly, it will automatically be wrapped in an Ok but we get type safety this approach instead of an IActionresult which can contain any content